app/src/main/java/com/dcrandroid/dialog/send/SendDialog.kt

```diff
@@ -125,9 +132,71 @@ class SendDialog(val fragmentActivity: FragmentActivity, dismissListener: Dialog
+
+            //fetch and save the selected source account
+            val selectedSourceAccount = amountHelper.selectedAccount
+            savedInstanceState?.putSerializable(Constants.SELECTED_SOURCE_ACCOUNT, selectedSourceAccount)
+
+            //fetch and save the selected destination account
+            val selectedDestAccount = destinationAddressCard.destinationAccountSpinner.selectedAccount
+            savedInstanceState?.putSerializable(Constants.SELECTED_DESTINATION_ACCOUNT, selectedDestAccount)
+
+            //save destination address state depending on if spinner or address input was selected
+            if(destinationAddressCard.isSendToAccount){
+                savedInstanceState?.putBoolean(Constants.SPINNER, true)
+            } else {
+                savedInstanceState?.putBoolean(Constants.SPINNER, false)
+            }
+    }
+
     override fun onResume() {
         super.onResume()
         destinationAddressCard.addressInputHelper.onResume()
+        if (savedInstanceState != null) {
+            //fetch saved sendMax Value
+            val max = savedInstanceState!!.getBoolean(Constants.SEND_MAX)
+            sendMax = max
+
+            //fetch saved destination address state
+            val spinner = savedInstanceState!!.getBoolean(Constants.SPINNER)
+
+            //fetch saved source and destination accounts
+            val selectedSourceAccount = savedInstanceState!!.getSerializable(Constants.SELECTED_SOURCE_ACCOUNT) as Account
+            val selectedDestAccount = savedInstanceState!!.getSerializable(Constants.SELECTED_DESTINATION_ACCOUNT) as Account
+
+            //Subtracted 1 to account for ArrayList counting
+            val selectedSourceAccountList = selectedSourceAccount.walletID - 1
+            val selectedDestAccountList = selectedDestAccount.walletID - 1
+
+            //Update UI based on previously selected source account
+            val sourceWalletID = multiWallet.openedWalletsList()[selectedSourceAccountList.toInt()]
+            val sourceWalletAcc = Account.from(sourceWalletID.getAccount(selectedSourceAccount.accountNumber))
+            sourceAccountSpinner.wallet = sourceWalletID
+            sourceAccountSpinner.selectedAccount = sourceWalletAcc
+            sourceAccountSpinner.selectedAccountChanged?.let { it1 -> it1(sourceAccountSpinner)}
+
+            //Update UI based on previously selected send to account
+            val destWalletID = multiWallet.openedWalletsList()[selectedDestAccountList.toInt()]
+            val destWalletAcc = Account.from(destWalletID.getAccount(selectedDestAccount.accountNumber))
+            destinationAddressCard.destinationAccountSpinner.wallet = destWalletID
+            destinationAddressCard.destinationAccountSpinner.selectedAccount = destWalletAcc
+            destinationAddressCard.destinationAccountSpinner.selectedAccountChanged?.let { it1 -> it1(destinationAddressCard.destinationAccountSpinner)}
```

Comment:
This entire operation is redundant and should be replaced with a function to set the selected account in `AccountCustomSpinner`.